pkg/pillar/nireconciler/genericitems/process.go

```diff
@@ -60,11 +61,22 @@ func startProcess(ctx context.Context, log *base.LogObject, cmd string, args []s
 		}
 		time.Sleep(1 * time.Second)
 	}
+	pid, _ := getProcessPid(log, pidFile)
+	log.Noticef("Started process %s %v with PID %d", cmd, args, pid)
 	return nil
 }
 
+func getPidOfExitedCmd(cmd *exec.Cmd) string {
+	pid := "?"
+	if cmd.ProcessState != nil && cmd.ProcessState.Pid() != 0 {
+		pid = fmt.Sprintf("%d", cmd.ProcessState.Pid())
+	}
+	return pid
+}
+
 func stopProcess(ctx context.Context, log *base.LogObject,
 	pidFile string, timeout time.Duration) error {
+	pid, _ := getProcessPid(log, pidFile)
```

Comment:
Odd to not check for error since in the future is someone changes the code below to use pid for the sendSignal insyead of pidFile, then on if the getProcessPid fails you'll be doing a kill 0 i.e., sending the signal to all processes in the callers process group.

Reply:
Good point - added error check.